### DISCUSSION TITLE
Use `Macroable` or allow adding custom attributes to `RouteRegistrar`

### DISCUSSION THREAD

**Discussion by bilogic** [laravel/framework#45557] [Ideas]
https://github.com/laravel/ideas/issues/741

If we add a new mixin/macro on `Illuminate\Routing\Route`, it cannot be accessed from `RouteRegistrar` because the `$allowedAttributes` are hardcoded. 

Since `Route` is macroable, it would be great if those macros can also be accessed from `RouteRegistrar`.

I found a need for this when trying to define an icon for each route, which can be turned into a menu/breadcrumbs.

**Comment by marinario** [comment]
@bilogic any work around for this?

**Reply by henzeb** [reply]
Closures. Can be bound to any object, anywhere anytime, and from there you can access the properties from RouteRegistrar. Just be sure you bind to the object and then use the context (actual class) the variable is in.
